src/l500/l500-color.h

```diff
@@ -61,15 +64,22 @@ namespace librealsense
             _state(sensor_state::CLOSED)
         {
         }
+        rs2_intrinsics get_raw_intrinsics( const uint32_t & width, const uint32_t & height ) const;
+        double read_temperature() const;
 
         rs2_intrinsics get_intrinsics( const stream_profile& profile ) const override;
-        
+
+
+
         // calibrated_sensor
         void override_intrinsics( rs2_intrinsics const& intr ) override;
         void override_extrinsics( rs2_extrinsics const& extr ) override;
         rs2_dsm_params get_dsm_params() const override;
         void override_dsm_params( rs2_dsm_params const & dsm_params ) override;
         void reset_calibration() override;
+        void set_k_thermal_intrinsics( rs2_intrinsics const & intr );
+        void reset_k_thermal_intrinsics(); 
+        algo::thermal_loop::l500::thermal_calibration_table get_thermal_table() const;
```

Comment:
Can we return a const &?

Reply:
I think its not safe , each time we will use the table we will need to check that the sensor still alive